### PR TITLE
FEAT(client, ui): Revamped server information dialog

### PR DESCRIPTION
**Pull request by Krzmbrzl**
The previous dialog was simply a message box with a bunch of HTML in
order to give a little structure to the contents. This approach is very
limiting in terms of UI flexibility though and therefore this commit
replaces the old HTML approach with a dedicated dialog class that uses
proper UI elements instead.

While doing so, the ordering and grouping of information was also
changed in order to make it more suitable for the every-day-user.

Fixes #4733

### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

-----

### Screenshots

Mumble_ServerInfo01
Mumble_ServerInfo02
Mumble_ServerInfo03

For reference this is the old window:
Old

### CODE REVIEW

**Comment by Krzmbrzl** [comment]
@SeerLite please let me know if you think this is sufficient or whether there is something that you think should be changed / improved?

**Comment by davidebeatrici** [comment]
I like it, however the `UDP (Voice)` tab should be hidden and the `TCP (Control)` one be called `TCP (Control + Voice)` when UDP is not active.

**Comment by Krzmbrzl** [comment]
I thought about something like this as well, but concluded that this will probably confuse users more (if they read somewhere to check their UDP statistics, then I think it is better to have that tab available that then just says that voice is currently sent through TCP as well) :point_up:

**Comment by davidebeatrici** [comment]
I agree.

File: src/mumble/ServerInformation.ui [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+         <width>40</width>
+         <height>20</height>
+        </size>
+       </property>
+      </spacer>
+     </item>
+     <item>
+      <widget class="QPushButton" name="viewCertButton">
+       <property name="text">
+        <string>&amp;View vertificate</string>
```
Comment (davidebeatrici) [inline src/mumble/ServerInformation.ui:626]:
```suggestion
        <string>&amp;View certificate</string>
```

Comment (Krzmbrzl) [reply src/mumble/ServerInformation.ui:626]: Haha whoopsie xD

**Comment by SeerLite** [comment]
Woah that's amazing. It looks a _lot_ better this way! I really should have let you do it from the start (as this was way too advanced for me). Sorry for forgetting about it :s

Thank you so much, all these changes are great. You managed to solve all the issues I had with the original window. This really makes Mumble look a lot more modern than before.

Only improvement I could think of was a small button next to the host and port fields to copy to clipboard, to make it easier to share with friends and contacts. But then I found out about #4847, which would make it as easy in a neater way. I'm not sure if there's any other use case for easy copying of host/port, but there's the idea.

Anyway I think it's perfect like this. Thank you again for everything

**Comment by Krzmbrzl** [comment]
> I really should have let you do it from the start (as this was way too advanced for me). Sorry for forgetting about it :s

No worries. In fact I think I would not have found the motivation to work on this if you hadn't created your PR ^^

> Only improvement I could think of was a small button next to the host and port fields to copy to clipboard, to make it easier to share with friends and contacts

You are actually able to select the respective texts and use Ctrl+C on it. Thus you can copy it to clipboard without a button. The only thing I could imagine is a button that lets you copy a complete "Mumble URL" (can be used when starting Mumble to directly connect to that server) as suggested by @Kissaki, but in the end I don't think it is really worth it :shrug: 

> Anyway I think it's perfect like this. Thank you again for everything

Great to hear that :+1: 
You're welcome :)

**Pull request merged**